### PR TITLE
md to sm changes by ChelseaSingla1

### LINKED ISSUE #102: [ENHANCEMENT] Refactoring of all the md: to sm:
### Description

We can aim for default classes (mobile), sm (tablet), lg (desktop if explicitly required), and lets refactor all the other classes if it were using md.

### Current Behavior

The current codebase uses the md: prefix for classes to define styling and behavior, presumably targeted at medium-sized devices. The specific styling associated with these classes may be tailored for devices with a medium screen size, such as tablets or devices in a certain resolution range.

### Proposed Enhancement

The proposed enhancement involves refactoring the existing codebase to improve class naming conventions and create a more consistent structure. The main changes include:

Replacing all instances of md: with sm: to better reflect the intended target, focusing on small to medium-sized devices.

### Use Case

The sm: classes, replacing the existing md: classes, would be employed for tablet-sized devices, providing a tailored layout and styling for this category.

### Blockers

_No response_

### Additional Information/ Attachments

_No response_

### PR DESCRIPTION
**Pull request by Chelseasingla1**
## Summary

This pull request is based on refactoring the md: to sm:

## Description

I made changes in all tsx files mainly changing md: to sm: wherever it was left.

## Issue(s) Addressed

Closes #102 

## Prerequisites

Yes i have followed all [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions).
- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Thank you for the PR @Chelseasingla1, some unintential changes i see, kindly address.

> Yes i have followed all [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions).

You have missed the 2nd and 3rd one 

You can refer to this PR , the title and the commit message
https://github.com/krishnaacharyaa/wanderlust/pull/179

But as this involves quite a few changes, i wouldn't tell you to recreate PR. 
Please obey from next time.

p.s: But if you are doing find and replace md: to sm: , kindly recreate PR. That would be appreciated actually

File: frontend/src/components/post-card.tsx [hunk 1]
```diff
@@ -10,11 +10,11 @@ export default function PostCard({ post, testId = 'postcard' }: { post: Post } &
...
     <div
-      className={`active:scale-click group w-full md:w-1/2 lg:w-1/3 xl:w-1/4`}
+      className={`active:scale-click group w-full sm:w-1/2 lg:w-1/3 xl:w-1/4`}
       data-testid={testId}
     >
       <div
-        className={`mb-4 cursor-pointer rounded-lg bg-light shadow-md dark:bg-dark-card ${'md:mr-8 md:mt-4'}`}
+        className={`mb-4 cursor-pointer rounded-lg bg-light shadow-sm dark:bg-dark-card ${'sm:mr-8 sm:mt-4'}`}
         onClick={() => navigate(`/details-page/${slug}/${post._id}`, { state: { post } })}
       >
```
[the remark this answers is not in the document]
Comment (Chelseasingla1) [reply frontend/src/components/post-card.tsx:19]: > Please revert back all such changes

@krishnaacharyaa so i need to make new pr request right?

[the remark this answers is not in the document]
Comment (krishnaacharyaa) [reply frontend/src/components/post-card.tsx:19]: Yes that would be better actually @Chelseasingla1, or other way of doing is just adding new commit with the reverted changes. Kindly follow the commit guidelines and change the PR title even as requested.

[the remark this answers is not in the document]
Comment (Chelseasingla1) [reply frontend/src/components/post-card.tsx:19]: > Yes that would be better actually @Chelseasingla1, or other way of doing is just adding new commit with the reverted changes. Kindly follow the commit guidelines and change the PR title even as requested.

@krishnaacharyaa okay sir

**Pull request closed without merging**